source/client/factories_impl.h

```diff
@@ -58,14 +61,30 @@ class OutputFormatterFactoryImpl : public OutputFormatterFactory {
 
 class RequestSourceFactoryImpl : public OptionBasedFactoryImpl, public RequestSourceFactory {
 public:
-  RequestSourceFactoryImpl(const Options& options);
+  RequestSourceFactoryImpl(const Options& options, Envoy::Api::Api& api);
   RequestSourcePtr create(const Envoy::Upstream::ClusterManagerPtr& cluster_manager,
                           Envoy::Event::Dispatcher& dispatcher, Envoy::Stats::Scope& scope,
                           absl::string_view service_cluster_name) const override;
 
+protected:
+  Envoy::Api::Api& api_;
+
 private:
   void setRequestHeader(Envoy::Http::RequestHeaderMap& header, absl::string_view key,
                         absl::string_view value) const;
+  /**
+   * Instantiates a RequestSource using a RequestSourcePluginFactory based on the plugin name in
+   * |config|, unpacking the plugin-specific config proto within |config|. Validates the config
+   * proto.
+   *
+   * @param config Proto containing plugin name and plugin-specific config proto.
```

Comment:
Not all parameters are documented, consider adding doc comments for `api` & `header`?